LDKNodeMonday/View Model/Home/OnboardingViewModel.swift

```diff
@@ -5,146 +5,39 @@
 //  Created by Matthew Ramsden on 1/4/24.
 //
 
+import Combine
 import LDKNode
 import SwiftUI
 
 class OnboardingViewModel: ObservableObject {
-    @Binding var appState: AppState
-    private let lightningClient: LightningNodeClient
-    private let keyClient: KeyClient
-    @Published var networkColor = Color.gray
+    let lightningClient: LightningNodeClient
+    @Published var networkSettingsViewModel: NetworkSettingsViewModel
+    @Binding var walletClient: WalletClient
     @Published var onboardingViewError: MondayError?
     @Published var seedPhrase: String = "" {
         didSet {
             updateSeedPhraseArray()
         }
     }
     @Published var seedPhraseArray: [String] = []
-    @Published var selectedNetwork: Network = .signet {
-        didSet {
-            do {
-                guard let server = availableEsploraServers.first else {
-                    // This should never happen, but if it does:
-                    fatalError(
-                        "Configuration error: No Esplora servers available for \(selectedNetwork)"
-                    )
-                }
-                self.selectedEsploraServer = server
-                try keyClient.saveNetwork(selectedNetwork.description)
-            } catch {
-                DispatchQueue.main.async {
-                    self.onboardingViewError = .init(
-                        title: "Error Selecting Network",
-                        detail: error.localizedDescription
-                    )
-                }
-            }
-        }
-    }
-    @Published var selectedEsploraServer: EsploraServer = EsploraServer.mutiny_signet
-    {
-        didSet {
-            do {
-                try keyClient.saveServerURL(selectedEsploraServer.url)
-            } catch {
-                DispatchQueue.main.async {
-                    self.onboardingViewError = .init(
-                        title: "Error Selecting Esplora",
-                        detail: error.localizedDescription
-                    )
-                }
-            }
-        }
-    }
-    var availableEsploraServers: [EsploraServer] {
-        switch selectedNetwork {
-        case .bitcoin:
-            return Constants.Config.EsploraServerURLNetwork.Bitcoin.allValues
-        case .testnet:
-            return Constants.Config.EsploraServerURLNetwork.Testnet.allValues
-        case .regtest:
-            return Constants.Config.EsploraServerURLNetwork.Regtest.allValues
-        case .signet:
-            return Constants.Config.EsploraServerURLNetwork.Signet.allValues
-        }
-    }
+    private var cancellables = Set<AnyCancellable>()
```

Comment:
Was `cancellables` + combine the only way to get this piece working right in the feature? Wondering just since its not the pattern we use in the codebase, but if its the only way to get this to work then obviously we should do it, but didn't know what the experience was liker when you used @ObservedObject or stuff like that on the view model here to trigger UI updates

Reply:
Good point.
I went back and looked at my previous PR to see how it was avoided there.
Short answer, I implemented a similar solution here which got rid of the need for Combine.

- Don't use a separate NetworkSettingsViewModel (which led to ObservableObjects observing each-other)